### DISCUSSION TITLE
What is the latest MAME version supported in macOS Catalina 10.15?

### DISCUSSION THREAD

**Discussion by wyatt-wong** [mamedev/discussions#116] [Q&A] [answered]
What is the latest MAME version supported in macOS Catalina 10.15?

**Comment by rb6502** [comment, marked as answer]
Current (0.267 soon) with this unofficial/unsupported patch: https://gist.github.com/rb6502/b3091421d0fd4f75c648359ee590ab9c

**Reply by wyatt-wong** [reply]
> Current (0.267 soon) with this unofficial/unsupported patch: https://gist.github.com/rb6502/b3091421d0fd4f75c648359ee590ab9c

Unfortunately, there is no binary built for macOS Catalina for the current version 0.266 or upcoming version 0.267 and I could NOT compile into binary from the latest source in macOS Catalina.

Would it be possible to provide a binary release for macOS Catalina ?

**Reply by galibert** [reply]
Did you try applying the patch before compiling?

**Reply by wyatt-wong** [reply]
> Did you try applying the patch before compiling?

You mean this patch can allow Xcode to build the source into binary under macOS Catalina 10.15 ?

https://gist.github.com/rb6502/b3091421d0fd4f75c648359ee590ab9c

**Reply by galibert** [reply]
That was what RB meant, yes.

**Reply by wyatt-wong** [reply]
> That was what RB meant, yes.

I downloaded the mame source from https://github.com/mamedev/mame and build the source under macOS 10.15.7 environment and I got the error below:

```
....
....
Compiling src/mame/atari/stmmu.cpp...
Compiling src/mame/atari/stvideo.cpp...
Compiling src/mame/atari/subs.cpp...
Compiling src/mame/atari/subs_a.cpp...
Compiling src/mame/atari/tank8.cpp...
Compiling src/mame/atari/tank8_a.cpp...
Compiling src/mame/atari/tempest.cpp...
Compiling src/mame/atari/thunderj.cpp...
Compiling src/mame/atari/tia.cpp...
Compiling src/mame/atari/tomcat.cpp...
Compiling src/mame/atari/toobin.cpp...
Compiling src/mame/atari/tourtabl.cpp...
Compiling src/mame/atari/triplhnt.cpp...
Compiling src/mame/atari/triplhnt_a.cpp...
Compiling src/mame/atari/tunhunt.cpp...
Compiling src/mame/atari/ultratnk.cpp...
Compiling src/mame/atari/videopin.cpp...
Compiling src/mame/atari/videopin_a.cpp...
Compiling src/mame/atari/vindictr.cpp...
Compiling src/mame/atari/wolfpack.cpp...
Compiling src/mame/atari/xybots.cpp...
Archiving libatari.a...
make: *** [makefile:1349: macosx_x64_clang] Error 2
```

**Reply by wyatt-wong** [reply]
After checking, the source file [3rdparty/bgfx/src/renderer_mtl.mm](https://github.com/mamedev/mame/blob/f0bf4f69a83c065965831fa515ccf639201558e1/3rdparty/bgfx/src/renderer_mtl.mm) need to be modified as mention below in order to be successfully compiled under macOS 10.15

https://gist.github.com/rb6502/b3091421d0fd4f75c648359ee590ab9c

So will this patch be officially made into the source code ?

**Comment by rb6502** [comment]
No.  Our downstream dependancies require macOS 11.0 and that forces us to move along with them.  For the time being that patch should continue to work with new MAME versions, but we can't guarantee that builds made that way will work 100% and we can't guarantee that the patch will continue to work.  I've already moved to Apple Silicon personally (it's much, much faster at compiling large C++ projects like MAME than my Intel MacBook Pro was) and many of macOS Sequoia's new features this fall will not work on Intel.
